src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -75,7 +80,12 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
             LogicalPlanFingerprint.Properties(Seq(Signature(signatureProvider.name, s)))))
 
         val coveringIndexProperties =
-          (hasLineageProperty(spark) ++ hasParquetAsSourceFormatProperty(spark, df)).toMap
+          (hasLineageProperty(spark) ++
+            hasParquetAsSourceFormatProperty(spark, df) ++
+            Hyperspace
+              .getContext(spark)
+              .sourceProviderManager
+              .versionHistoryProperties(versionId, relations.head, prevIndexProperties())).toMap
```

Comment:
Instead of having this API, how about exposing something generic like `def enrichRelationProperties(relation: Relation, prop: Map[String, String]): Map[String, String]`? ' You will pack the `versionId` as an entry in the map; the default implementation will just return the given `prop`.

Reply:
in FileBasedRelation? but in `optimizeAction`, we don't construct source df.. 
In sourceProvider, yes I can make it more generic.

Reply:
yes, this will be in source provider.